### ISSUE TITLE
Time zone change causes Dailies to become offset by 1 day

### ISSUE THREAD

**Issue by gpunktschmitz** [HabitRPG/habitica#8182]
General Info
  * UUID: 7894c234-4aaa-48dd-a244-adb5ef975ca3
  * Browser: Opera 41.0.2353.46
  * OS: Linux Mint 17.3 Rosa

### Description
I've set up a Daily to repeat every 7 days and the Start Date "19/08/2016" (which was a Friday). Ever since it poped up every Friday. Today however (it is Thursday) the Daily shows up under the "Due" filter.

I could set a new Start Date an reschedule this item .. but wondered if this bug could be fixed somehow.

This bug can be reproduced by creating a new Daily and set:
-Start Date "19/08/2016"
-Repeat "Every X Days"
-Repeat every "7 Days"

**Comment by Alys** [comment]
Did you go through a time zone change recently? If so, the Daily is still set to repeat every 7 days, it's just become offset by one day due to the zone change. If you change the Start Date, it will appear on the correct day again.

**Comment by gpunktschmitz** [comment]
yes I did go through a time zone change and rescheduled the Daily.
so it seems like a common issue ...

**Comment by JenniferMartin** [comment]
I'm leaving my issue here because it seems to be related, although I'm not sure it's entirely the same thing.

I have a number of dailies which are set to be due every 28 days. Two of them should have been due today. However, when I logged in today, neither were due, their streaks were missing, and I did not have the perfect day bonus I should have had from yesterday. I checked the data display tool, and sure enough, they were shown as having been missed yesterday--which was odd, as they were not showing as due yesterday during the day. However, I figured this was probably due to some odd interaction between my custom day start of 4 AM and the fact that daylight savings time ended last night at 2 AM (although all my "due on day of week" dailies are fine). So for the two problem dailies, I tried resetting their "start on" date for today, figuring that would get them back onto the right days. That didn't work--they were still not showing as due today. I also tried setting them for tomorrow and for yesterday, just in case the day indicator had somehow gotten really scrambled. No dice. I tried setting the "start on" date to 28 days ago. Still nothing. I finally tried setting the "start on" date to 27 days ago, and that did work to get them to show as due. I will be keeping an eye on them, as well as on the other similar dailies due other weeks, to see how things play out with due dates down the road.

However, I still find it odd that setting the "start on" date to today did not get them to show as due today, which is the primary reason I'm reporting it.

My user ID is 154802ac-9567-4916-ac66-79ea5ba81ec7, in case you want to take a look at the back end of what happened.

**Comment by Alys** [comment]
@JenniferMartin 

> this was probably due to some odd interaction between my custom day start of 4 AM and the fact that daylight savings time ended last night at 2 AM

I agree.

> (although all my "due on day of week" dailies are fine)

I believe the mis-scheduling problem that happens when DST changes can depend on the (hidden and inaccessible) time portion of the start date. Start dates are stored as UTC times and converted to your local time zone when scheduling is calculated. For certain UTC times, a different local time zone can be enough to shift the Daily to the wrong day. That could explain why some Dailies were mis-scheduled but not all (the "due on day of week" might be coincidental).

Were you editing the start dates when the time in your local area was after midnight but before 4am? If so, the failure could have been due to this issue: https://github.com/HabitRPG/habitica/issues/5525

**Comment by JenniferMartin** [comment]
@Alys 
I had not edited the dailies in question for several months, so I don't remember what time I had most recently made changes. Today's attempts to get them to display as due were made around 9:30 AM local time (EST).

**Comment by lemoness** [comment]
Is this still happening?

**Comment by JenniferMartin** [comment]
I haven't had the issue crop up again once I got the dailies sorted out in November, but we also haven't had a time change since then, so I cannot say for sure whether the problem still exists.

**Comment by Alys** [comment]
The problem I described above with time zone changes causing a Daily to become offset by one day still happens. I thought we had a more specific issue logged for it but I can't find it.
